### ISSUE TITLE
Getting ban

### ISSUE THREAD

**Issue by almozayaf** [ccloli/E-Hentai-Downloader#133]
I got banned from E-Hentai for two days now, because of this script, I'm trying to get 300 point a day only for now just not to get banned. any thing else I can do

**Comment by ccloli** [comment]
If you mean your IP got banned, as https://ehwiki.org/wiki/Technical_Issues#Ban says, you can do nothing but wait. Or you can do some tricks like reconnecting your network to your ISP (redial PPPoE) if you're using a dynamic IP address, so that you can get a new IP, or you can try using a VPN or some other proxy tools.

Maybe next time you can try reducing `Download _____ images at the same time`, and setting `Delay _____ second(s) before requesting next image` to avoid banning.

**Comment by almozayaf** [comment]
I made it downloading 2 Images and 3 second delay hope it worked

**Comment by AtotehZ** [comment]
I got banned as well now.
24h on Saturday
3 days starting Tuesday.

I counted the pages I'd visited, it was less than 500, so it must be loading a lot of pages on its own.
I'm paying for a static IP and when I asked them they told me it wasn't shared with anyone.
I also don't have any other scripts running.

It's curious that I got banned now.. I'd actually forgotten the script was on, it has just updated itself in the background. I've had it running for a very long time.

I'll be disabling the script for now, but would very much like to know how I can reduce the pageloads by as much as possible.

If I end up getting banned a 3rd time it can't be with a script on. I couldn't say for sure I hadn't done what they said then.

**Comment by ccloli** [comment]
@AtotehZ  Hi, sorry to hear that. If it's cleared that your IP is static and not shared with others, a good news is that you owns your image limits, but bad news is that you can't reset your limit by reconnecting your network to obtain a new IP address.

Here is a check list that may help you to find the reason:

> tl;dr, if you think you're not doing anything wrong, it might be the third reason that the script may send multiple requests to get your image limits at a small chance, and it'll be optimized in next version.

- Are you downloading multiple galleries or too many images at the same time these days?
     If so, please consider reduce download queue and increase delay interval by reducing `Download _____ images at the same time`, and setting `Delay _____ second(s) before requesting next image`.
- Are you leaving the script download on the background but it continuous fail to download?
     If you've enabled `Retry automatically when images download failed` but left it in background tabs, it may recheck infinity times and even cost all your image limits. You can check https://e-hentai.org/home.php to see if your image limits are significantly larger than normal.
- Are you opening multiple gallery pages in background?
    I reviewed some older issues just now, and remembered that Chrome has some optimization on `setTimeout()`, which reminds me that the script will check your image limits at every minutes on each tab. The script has a throttle limit to make sure that only 1 tab will query the image limits at each minute, so it shouldn't send a bunch of requests. However as Chrome has the optimization, there's a small chance that the throttle function may not work.
    As my normal usage, leave about ~20 tabs in background shouldn't be a problem. Not sure if the 500 pages means you opened 500 tabs on the background, but if so, it might be the reason. Anyway I'll try to fix it in next version, for now you can try enabling `Disable requesting and showing image limits`.
- Are you having multiple devices browsing E-Hentai at the same time?
    All your devices in your network are using the same limit, as it based on your IPs.
- Are you sharing your account with others?
    There's no report showing that sharing account will ban your IP, but the account might be blocked.

And if you've privacy concerns, the script won't grab any pages without your action, and there is no background fetching task or sharing network.

**Comment by AtotehZ** [comment]
I'm not sure it had anything to do with the image limit, but I can't confirm it as that would require me to have access on the banned IP.

This is the message I get on my IP:
"Your IP address has been temporarily banned for excessive pageloads which indicates that you are using automated mirroring/harvesting software. The ban expires in 2 days and 23 hours" <- about 30 hours left now.

- I have not downloaded any galleries at all on these days.
- I often have a lot of tabs in the background and 500 pages were the pages I visited manually.
- I do have multiple devices I use E-hentai with, but only my main PC has any scripts installed.
- I am not account sharing, but thanks for reminding me about that. If it actually is my account being abused I can just change the password. It would be weird that they take action against my IP and not my account then though. I can still access my account on other IPs.

**Comment by ccloli** [comment]
> "Your IP address has been temporarily banned for excessive pageloads which indicates that you are using automated mirroring/harvesting software. The ban expires in 2 days and 23 hours" <- about 30 hours left now.

Is it still counting down? As you said your IP has been blocked for 3 days, but looks like the time doesn't change. The countdown should be up to date, at least when I ran script for another project (the QPM or requests per minute is around 40, and ran for about 3 and a half hours), my IP was banned for an hour, and the time is up to date:

```
Your IP address has been temporarily banned for excessive pageloads which indicates that you are using automated mirroring/harvesting software. The ban expires in 53 minutes and 27 seconds
......
Your IP address has been temporarily banned for excessive pageloads which indicates that you are using automated mirroring/harvesting software. The ban expires in 50 minutes and 10 seconds
......
Your IP address has been temporarily banned for excessive pageloads which indicates that you are using automated mirroring/harvesting software. The ban expires in 46 minutes and 21 seconds
......
Your IP address has been temporarily banned for excessive pageloads which indicates that you are using automated mirroring/harvesting software. The ban expires in 4 seconds
```

**Unless** E-Hentai restricts the flow control limit, I think opening ~30 tabs of E-Hentai gallery page and in worst case all of them querys image limits at the same time, shouldn't trigger the IP ban.

> I do have multiple devices I use E-hentai with, but only my main PC has any scripts installed.

Not sure for third party apps, but maybe they have feature like preload images, and if the app requests too fast can also makes the server blocks your IP. 

Or if we discuss in conspiracy theory, there is a **very very small** chance and closes to never happen, that someone are crawling E-Hentai but treats his IP as yours. In technically, it may works with setting `X-Forwarded-For` and `X-Real-IP` HTTP headers, and if the server accepts these headers and trust them, then all of his requests will count to you! lol

**Comment by ccloli** [comment]
![image](https://user-images.githubusercontent.com/8115912/121574761-2ebd7380-ca59-11eb-8454-6d81374099da.png)

Well, I think that Chrome's optimization do matters. After leaving these tabs in background for an hour, these tabs request viewing limits at the same time, though they're not opened at the same time. 

A bugfix version will be updated tomorrow, and for now it's better to `Disable requesting and showing image limits`. If you're still being blocked and leave those tabs in background, and cannot turn on the option, I recommend to disable the script and restart the browser, since the tabs opened before the script disabled are still sending requests.

**Comment by AtotehZ** [comment]
That sounds good.
Delete the old version to be sure.

I didn't display an updated counter because I was on a VPN when talking to you. I copy/pasted it from a thread on E-hentai.

This much left:
"Your IP address has been temporarily banned for excessive pageloads which indicates that you are using automated mirroring/harvesting software. The ban expires in 5 hours and 46 minutes"

**Comment by AtotehZ** [comment]
I just tried this:
https://www.tenforums.com/tutorials/146276-how-enable-disable-tab-freezing-google-chrome.html
For someone like me who has 500 tabs open at a time I think it might be a good thing. I'd rather reload a page than them running in the background.

**Comment by ccloli** [comment]
v1.33.2 is released just now, when you leave tabs in background, they won't querying image limits until you switch them in foreground.

> For someone like me who has 500 tabs open at a time I think it might be a good thing. I'd rather reload a page than them running in the background.

I remembered this feature came a few years ago and named as discard tabs, and you can manage your tabs or kill them in `chrome://discards/`. It'll unload the tabs running in background and inactive for some time when the system memory is almost used out. It's useful when I've around 1000 tabs opened without breaking the system. Looks like the experimental feature you linked seems to be the same thing but uses a more radical algorithm.
